src/components/header/ProfileThreadHeaderContextMenu.js

```diff
@@ -47,11 +56,34 @@ class ProfileThreadHeaderContextMenu extends PureComponent<Props> {
     }
   }
 
+  _isolateThread = () => {
+    const { isolateThread, rightClickedThreadIndex } = this.props;
+    isolateThread(rightClickedThreadIndex);
+  };
+
   render() {
-    const { threads, threadOrder, hiddenThreads } = this.props;
+    const {
+      threads,
+      threadOrder,
+      hiddenThreads,
+      rightClickedThreadIndex,
+    } = this.props;
+
+    const clickedThreadName = getFriendlyThreadName(
+      threads,
+      threads[rightClickedThreadIndex]
+    );
 
     return (
       <ContextMenu id={'ProfileThreadHeaderContextMenu'}>
+        {hiddenThreads.length === threads.length - 1
+          ? null
+          : <div>
+              <MenuItem onClick={this._isolateThread}>
+                Only show: {`"${clickedThreadName}"`}
```

Comment:
I've mixed feelings about this label. I'm used to Firefox "close other tabs" so I think I'd prefer "hide other threads" here. I think this convey the intention better because this is an action.

Although if we don't have the thread name anywhere it's difficult to know which one will be displayed...

Ideally, we'd have thread name at the top of the context menu as a title, and have "hide other threads" as label for this action. However I understand this is more work than what's needed so let's keep it like this for now.

Reply:
I agree it's awkward.